framework/src/utils/MooseUtils.C

```diff
@@ -578,6 +578,40 @@ getDefaultExecFlagEnum()
                               EXEC_CUSTOM);
   return exec_enum;
 }
+
+int
+stringToInteger(const std::string & input, bool throw_on_failure)
+{
+  int output;            // return value
+  std::size_t count = 0; // number of characters converted with stoi
+
+  // lambda helper for throwing exception or mooseError
+  auto error_handler = [](const std::string & input, bool throw_on_failure) {
+    std::string msg = "Failed to convert '" + input + "' to an int.";
+    if (throw_on_failure)
+      throw std::invalid_argument(msg);
+    else
+      mooseError(msg);
+  };
+
+  // Attempt to use std::stoi, if it fails throw or produce a mooseError
+  try
+  {
+    output = std::stoi(input, &count);
```

Comment:
Right after this line, I'd do the check on size versus count (currently at the bottom of this routine). If that check fails, throw an invalid_argument. It's fine to test count here and you know there's no error because you'd have already left the try block if there was a problem. By doing this you can localize count and you can also drop the lambda since you can just print the message or fail directly in the catch block since both types of error cases will be sent to the same location (stoi will throw on failure and your size check will also throw when different). That'll cut this routine by more than 50% and make it easier to follow and understand.

Reply:
You are right, not sure why I didn't think of that.